### ISSUE TITLE
Segmentation segfaults on some meshes

### ISSUE THREAD

**Issue by mpowelson** [mpowelson/noether#1]
This has shown up for several meshes. It seems like it occurs on the last mesh. I'm placing this here to keep track of my notes on this issue. I have seen it occur in my segmentation_devel branch and have not confirmed it occurs in master.

Backtrace:

```
Thread 6 (Thread 0x7f663e416700 (LWP 31338)):
#0  0x00007f66594731cb in vtkAOSDataArrayTemplate<int>::GetValue (this=0x7f6629bf6ed0, valueIdx=697996624) at /home/mpowelson/workspaces/tpp/ext/VTK/Common/Core/vtkAOSDataArrayTemplate.h:64
No locals.
#1  0x00007f66586f98cd in vtkCellTypes::GetCellLocation (this=0x7f6629bf7020, cellId=697996624) at /home/mpowelson/workspaces/tpp/ext/VTK/Common/DataModel/vtkCellTypes.h:76
No locals.
#2  0x00007f66558cfad2 in vtkPolyData::GetCell (this=0x7f6629c0fb60, cellId=140076761388368, cell=0x7f661366df50) at /home/mpowelson/workspaces/tpp/ext/VTK/Common/DataModel/vtkPolyData.cxx:353
        i = 1433627731
        loc = 32614
        pts = 0x0
        numPts = 16
        type = 101 'e'
        x = {6.9207116825842438e-310, 6.9206904026480296e-310, 1.5035405734240815e-319}
#3  0x00007f66558cffd3 in vtkPolyData::CopyCells (this=0x7f661014d630, pd=0x7f6629c0fb60, idList=0x7f6629c0ff70, locator=0x0) at /home/mpowelson/workspaces/tpp/ext/VTK/Common/DataModel/vtkPolyData.cxx:450
        cellId = 30432
        ptId = 9
        newId = 20859
        newCellId = 30431
        locatorPtId = 140076763906240
        numPts = 70114
        numCellPts = 3
        i = 3
        newPoints = 0x7f661015eab0
        pointMap = 0x7f661366deb0
        cellPts = 0x7f661366e3f0
        newCellPts = 0x7f661366df00
        cell = 0x7f661366df50
        x = {3.6741104125976562, 1.0117188692092896, -0.62093567848205566}
        outPD = 0x7f661014d960
        outCD = 0x7f661014da60
#4  0x00007f665ab6fc4c in mesh_segmenter::MeshSegmenter::getMeshSegments (this=0x7f663e415720) at /home/mpowelson/workspaces/tpp_working/src/noether/mesh_segmenter/src/mesh_segmenter.cpp:52
        mesh = {<vtkSmartPointerBase> = {Object = 0x7f661014d630}, <No data fields>}
        i = 43
        input_copy = {<vtkSmartPointerBase> = {Object = 0x7f6629c0fb60}, <No data fields>}
        meshes = {<std::_Vector_base<vtkSmartPointer<vtkPolyData>, std::allocator<vtkSmartPointer<vtkPolyData> > >> = {_M_impl = {<std::allocator<vtkSmartPointer<vtkPolyData> >> = {<__gnu_cxx::new_allocator<vtkSmartPointer<vtkPolyData> >> = {<No data fields>}, <No data fields>}, _M_start = 0x7f661019ab30, _M_finish = 0x7f661019ac88, _M_end_of_storage = 0x7f661019ad30}}, <No data fields>}
#5  0x0000000000481784 in SegmentationAction::executeCB (this=0x7fff06c46960, goal=...) at /home/mpowelson/workspaces/tpp_working/src/noether/noether/src/segmentation_server.cpp:126
        curvature_threshold = 0.29999999999999999
        min_cluster_size = 500
        max_cluster_size = 1000000
        mesh = {<vtkSmartPointerBase> = {Object = 0x7f6628bd7f80}, <No data fields>}
        input_pcl_mesh = {header = {seq = 0, stamp = 0, frame_id = {static npos = 18446744073709551615, _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>}, _M_p = 0x7f663e4158b0 ""}, _M_string_length = 0, {_M_local_buf = "\000XA>f\177\000\000\340\023\371\001\000\000\000", _M_allocated_capacity = 140077107861504}}}, cloud = {header = {seq = 0, stamp = 0, frame_id = {static npos = 18446744073709551615, _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>}, _M_p = 0x7f663e4158e0 ""}, _M_string_length = 0, {_M_local_buf = "\000YA>\377\377\377\377PZA>f\177\000", _M_allocated_capacity = 18446744070459054336}}}, height = 1, width = 70114, fields = {<std::_Vector_base<pcl::PCLPointField, std::allocator<pcl::PCLPointField> >> = {_M_impl = {<std::allocator<pcl::PCLPointField>> = {<__gnu_cxx::new_allocator<pcl::PCLPointField>> = {<No data fields>}, <No data fields>}, _M_start = 0x7f6628000ee0, _M_finish = 0x7f6628000f70, _M_end_of_storage = 0x7f6628000f70}}, <No data fields>}, is_bigendian = 0 '\000', point_step = 12, row_step = 841368, data = {<std::_Vector_base<unsigned char, std::allocator<unsigned char> >> = {_M_impl = {<std::allocator<unsigned char>> = {<__gnu_cxx::new_allocator<unsigned char>> = {<No data fields>}, <No data fields>}, _M_start = 0x7f6628002420 "% k@\206\222\201?\205J\037\277d\036k@'\216\201?S?\037\277\364\030k@F\221\201?RF\037\277U\026k@\236\216\201?;:\037\277`\034k@[\212\201?)/\037\277c%k@a\204\201?\031\066\037\277\206$k@\030\200\201?\276\365\036\277\320%k@r}\201?/\016\037\277\337\035k@\n\215\201?\215;\037\277\240$k@\001\200\201?\244\365\036\277\065\033k@è?q#\037\277S%k@\231~\201?_\031\037\277\236&k@\306|\201? \003\037\277\265\023k@\221\215\201?(-\037\277\313:k@:Ç?\343$\037\277\066\067k@tÊ?~'\037\277\037%k@Ô\201?K'\037\277^\037k@$Þ?\217\067\037\277\060\021k@\033\216\201?\235\037\037\277S\017k@\245\217\201?\344\024\037\277\376*k@\277Ï?*?\037\277\207\031k@t\210\201?f\v\037\277\063\032k@I\210\201?\310\026\037\277\225\071k@SÆ?\362.\037\277\256\066k@\222\302\201?LB\037\277\306*k@`Ë?\256H\037\277\207!k@hØ?\260F\037\277\022\070k@\320Ä?*9\037\277\347#k@\327Ç?\247]\037\277\061\065k@h\277\201?\027K\037\277}*k@3Ç?\246O\037\277\022#k@\322Ð?\240S\037\277\257\063k@\221\273\201?\254R\037\277A2k@v\267\201?|X\037\277\341(k@\241\270\201?\313]\037\277\365)k@e\301\201?\351V\037\277\250(k@\024\267\201?\225^\037\277\363#k@\361\275\201?\030d\037\277\062\060k@\373\260\201?^^\037\277*,k@Æ¢\201?;a\037\277&.k@\375\251\201?Pa\037\277\201'k@æ¯?\266`\037\277\070#k@\261\263\201?\254f\037\277\306!k@\262\251\201?Pe\037\277h'k@\\\257\201?\305`\037\277\231%k@\034\246\201?\237_\037\277%&k@\264\250\201?a`\037\277\350\037k@k\241\201?\260`\037\277O*k@\236\233\201?\016^\037\277\235$k@Æ¡\201?\200]\037\277\276#k@H\236\201?\353Z\037\277\065'k@\206\216\201?\311N\037\277\242(k@Ë\201?\315W\037\277\364\"k@\\\233\201?\022X\037\277\364!k@\365\227\201?\332S\037\277}\033k@n\225\201?\377P\037\277\326\035k@â?\347Y\037\277\032&k@\376\210\201?iC\037\277:!k@\251\225\201?\\P\037\277\203\037k@Ø\201?\274F\037\277\330\025r@\253zj?\210\314\036\277*-r@\247\236j?\220\315\035\277'6r@\304Tj?\004\276\035\277\262\fr@\251\365j?\372\320\036\277l\020r@\024\267j?\235\314\036\277\351^S@\376\244\251?R\003\220\276\362vS@\a\214\251?2á¾}^S@X\215\251?sæ¾/pS@Bz\251?[\274\220\276\320TS@\036\225\251?\275y\222\276H0S@Ä\251?\374\n\226\276_\034U@\220\214\252?\217F\303\061\241&U@\340i\252?\247\236\267\262\222!S@f[\251?\245I\234\276uYS@\251\201\251?\"F\223\276\221!S@g[\251?\244I\234\276\253\017S@\222b\251?\323Þ\276\215\351R@\037j\251?[\001\241\276\334QS@S\215\251?5h\223\276\217MS@x\204\251?â\276\306\066S@)U\251?\336\003\232\276\377'S@\232K\251?\365O\234\276\026>S@\343Y\251?\257Ý\276\314\364R@7Q\251?\331\352\241\276\256\024S@xY\251?\246ä¾\277\033S@\371\065\251?\210\302\236\276\316\371R@\033H\251?\317\360\241\276\342\333R@\036A\251?'\224\245\276\336\004S@\202U\251?\fè¾\177\bS@\347Y\251?[:\237\276K\372R@\302T\251?@\031\241\276\\\001S@IU\251?\322M\240\276\215\351R@\017j\251?\356\002\241\276\037\374R@\303]\251?]G\240\276\321\377R@\241\061\251?\312b\242\276g\371R@qA\251?A\\\242\276g\371R@pA\251?B\\\242\276\253\363R@\211\060\251?Mß£\276\270\323R@\025)\251?\356Ù§\276\227\360R@\312'\251?k\240\244\276&\351R@\372\"\251?\355\265\245\276\267\323R@\026)\251?\355Ù§\276i\307R@\211'\251?\242B\251\276\225\272R@\216'\251?\373\227\252\276k\302R@\250\060\251?i<\251\276\036\332R@B\031\251?\251\340\247\276\n\001S@\003\t\251?]\237\243\276\354\256R@T\375\250?\354\362\255\276\237\266R@\361\"\251?\261E\253\276C\313R@\357\001\251?4\205\252\276D\256R@\242#\251?\266\022\254\276\226\235R@\004\071\251?\345\372\253\276\241\256R@\263+\251?+\203\253\276\223\245R@\005\036\251?\033H\255\276Ú³R@3#\251?\020\212\253\276\026\241R@%\373\250?\225\220\257\276&\235R@\327\361\250?ob\260\276\211\250R@-\r\251?\371ë­¾\217\252R@\343\024\251?xN\255\276\212\250R@+\r\251?\371ë­¾\242\225R@\372\354\250?Vg\261\276{\206R@-\343\250?\336p\263\276\033\200R@\a\363\250?\272i\263\276\033\200R@\b\363\250?\271i\263\276\230\235R@\026\071\251?\030\371\253\276jLR@\247\004\251?\200\361\266\276\220tR@\372\361\250?\200\240\264\276\024gR@\224\361\250?\305\365\265\276\004]R@\034\357\250?\311\f\267\276\337`R@\217\355\250?\266Æ¶\276fLR@\223\004\251?\177\363\266\276\310bR@\277\354\250?\240\243\266\276:XR@CÅ¨?È\271\276\337QR@ Õ¨?m|\271\276\223HR@\347\301\250?\204B\273\276\264CR@\375\267\250?\020%\274\276\006OR@\340\225\250?\313ë»¾0<R@\037\263\250?@\031\275\276\v-R@R\251\250?d\001\277\276\227oR@ \373\250?â´\276ÛR@\301\f\251?\217\361\260\276\262&R@5\271\250?\327\371\276\276g\354Q@\rV\250?X\022È¾\274\373Q@j\211\250?\201\024Å¾<\253R@\206Ñ¨?%Å¯\276\037\354Q@Q\177\250?\203\366Æ¾\376\327Q@\026\231\250?ß¬Ç¾D\344Q@;z\250?d\347Ç¾}\307Q@\232{\250?j\211Ê¾\322\332Q@\342z\250?w\310È¾\276PR@/\347\250?\025\246\270\276\315tR@\005Ê¨?\316G\266\276~\252Q@\226w\250?!?Í¾]\232Q@~\221\250?\265\343Ì¾L\252Q@gu\250?\301^Í¾\344\265Q@xN\250?\231\314Í¾\265UR@\aÞ¨?Ö¬\270\276\247\243Q@(s\250?4\nÎ¾3\233Q@[e\250?[hÏ¾\336QR@!Õ¨?l|\271\276\327~Q@\220\070\250?\205\250Ó¾\300\207Q@T>\250?0\247Ò¾\177bQ@8:\250?\222\027Ö¾\210uQ@N9\250?\234xÔ¾\200bQ@6:\250?\223\027Ö¾\312hQ@F*\250?\357\037Ö¾\371AQ@\330?\250?\301pØ¾\262&R@4\271\250?\330\371\276\276\035\034R@Æ¸\250?\211\376\277\276*\027R@\362\301\250?\212\367\277\276\071\064R@\277Ô¨?gO\274\276\376\365Q@\275Ì¨?\004\354\301\276\253EQ@A4\250?H\272Ø¾\033GQ@?7\250?KyØ¾\260>Q@\314\061\250?\231eÙ¾\222\071Q@\362R\250?i\327\327\276\217\065Q@\212#\250?\331\303Ú¾\250\060Q@\304,\250?\374\273Ú¾\213\006R@1\267\250?\274\006Â¾X\030R@3\216\250?\223\nÂ¾u\035Q@y\371\247?\323U\336\276A\373Q@\205\243\250?\022\vÄ¾n\023Q@\371\362\247?\207g\337\276U\nQ@\333\363\247?\225&à¾º\367P@\v\365\247?t\243\341\276Q\354P@~", _M_finish = 0x7f66280cfab8 "%", _M_end_of_storage = 0x7f66280cfab8 "%"}}, <No data fields>}, is_dense = 1 '\001'}, polygons = {<std::_Vector_base<pcl::Vertices, std::allocator<pcl::Vertices> >> = {_M_impl = {<std::allocator<pcl::Vertices>> = {<__gnu_cxx::new_allocator<pcl::Vertices>> = {<No data fields>}, <No data fields>}, _M_start = 0x7f663cb9f010, _M_finish = 0x7f663ce7d8f0, _M_end_of_storage = 0x7f663ce7d8f0}}, <No data fields>}}
        __PRETTY_FUNCTION__ = "void SegmentationAction::executeCB(const SegmentGoalConstPtr&)"
        mesh_in = {<vtkSmartPointerBase> = {Object = 0x7f6628bd7f80}, <No data fields>}
        mesh_cleaned = {<vtkSmartPointerBase> = {Object = 0x0}, <No data fields>}
        mesh_filtered1 = {<vtkSmartPointerBase> = {Object = 0x7f6628c73e20}, <No data fields>}
        mesh_filtered2 = {<vtkSmartPointerBase> = {Object = 0x7f6628c777f0}, <No data fields>}
        segmenter = {min_cluster_size_ = 500, max_cluster_size_ = 1000000, curvature_threshold_ = 0.29999999999999999, input_mesh_ = {<vtkSmartPointerBase> = {Object = 0x7f6628c777f0}, <No data fields>}, triangle_filter_ = {<vtkSmartPointerBase> = {Object = 0x7f662897e870}, <No data fields>}, included_indices_ = {<std::_Vector_base<vtkSmartPointer<vtkIdList>, std::allocator<vtkSmartPointer<vtkIdList> > >> = {_M_impl = {<std::allocator<vtkSmartPointer<vtkIdList> >> = {<__gnu_cxx::new_allocator<vtkSmartPointer<vtkIdList> >> = {<No data fields>}, <No data fields>}, _M_start = 0x7f6629bfc2e0, _M_finish = 0x7f6629bfc440, _M_end_of_storage = 0x7f6629bfc4e0}}, <No data fields>}}
        viz = {viewer_ = {renWin_ = 0x7f6628c74520, renderer_ = 0x7f6628c73900, iren_ = 0x7f6628bd86b0, mouse_interactor_ = 0x7f6628b3d4d0, actors_ = {<std::_Vector_base<vtkSmartPointer<vtkActor>, std::allocator<vtkSmartPointer<vtkActor> > >> = {_M_impl = {<std::allocator<vtkSmartPointer<vtkActor> >> = {<__gnu_cxx::new_allocator<vtkSmartPointer<vtkActor> >> = {<No data fields>}, <No data fields>}, _M_start = 0x7f6628b3d190, _M_finish = 0x7f6628b3d198, _M_end_of_storage = 0x7f6628b3d198}}, <No data fields>}, poly_mappers_ = {<std::_Vector_base<vtkSmartPointer<vtkPolyDataMapper>, std::allocator<vtkSmartPointer<vtkPolyDataMapper> > >> = {_M_impl = {<std::allocator<vtkSmartPointer<vtkPolyDataMapper> >> = {<__gnu_cxx::new_allocator<vtkSmartPointer<vtkPolyDataMapper> >> = {<No data fields>}, <No data fields>}, _M_start = 0x7f6628b3ce60, _M_finish = 0x7f6628b3ce68, _M_end_of_storage = 0x7f6628b3ce68}}, <No data fields>}}}
        tmp = {<std::_Vector_base<vtkSmartPointer<vtkPolyData>, std::allocator<vtkSmartPointer<vtkPolyData> > >> = {_M_impl = {<std::allocator<vtkSmartPointer<vtkPolyData> >> = {<__gnu_cxx::new_allocator<vtkSmartPointer<vtkPolyData> >> = {<No data fields>}, <No data fields>}, _M_start = 0x7f6628b383d0, _M_finish = 0x7f6628b383d8, _M_end_of_storage = 0x7f6628b383d8}}, <No data fields>}
        tStart = {<ros::TimeBase<ros::Time, ros::Duration>> = {sec = 1543957274, nsec = 79761180}, <No data fields>}
        segmented_meshes = {<std::_Vector_base<vtkSmartPointer<vtkPolyData>, std::allocator<vtkSmartPointer<vtkPolyData> > >> = {_M_impl = {<std::allocator<vtkSmartPointer<vtkPolyData> >> = {<__gnu_cxx::new_allocator<vtkSmartPointer<vtkPolyData> >> = {<No data fields>}, <No data fields>}, _M_start = 0x7f661019ab30, _M_finish = 0x7f661019ac88, _M_end_of_storage = 0x7f661019ad30}}, <No data fields>}
        pcl_mesh_msgs = {<std::_Vector_base<pcl_msgs::PolygonMesh_<std::allocator<void> >, std::allocator<pcl_msgs::PolygonMesh_<std::allocator<void> > > >> = {_M_impl = {<std::allocator<pcl_msgs::PolygonMesh_<std::allocator<void> > >> = {<__gnu_cxx::new_allocator<pcl_msgs::PolygonMesh_<std::allocator<void> > >> = {<No data fields>}, <No data fields>}, _M_start = 0x6e6700 <actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > >::setAborted(noether_msgs::SegmentResult_<std::allocator<void> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::__rosconsole_define_location__loc>, _M_finish = 0x47e10e <boost::mutex::unlock()+32>, _M_end_of_storage = 0x7f6628000a80}}, <No data fields>}
        success = false
#6  0x00000000004a79b4 in boost::_mfi::mf1<void, SegmentationAction, boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> const&>::operator() (this=0x7fff06c46bd8, p=0x7fff06c46960, a1=...) at /usr/include/boost/bind/mem_fn_template.hpp:165
No locals.
#7  0x00000000004a4a65 in boost::_bi::list2<boost::_bi::value<SegmentationAction*>, boost::arg<1> >::operator()<boost::_mfi::mf1<void, SegmentationAction, boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> const&>, boost::_bi::list1<boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> const&> > (this=0x7fff06c46be8, f=..., a=...) at /usr/include/boost/bind/bind.hpp:313
No locals.
#8  0x00000000004a0e11 in boost::_bi::bind_t<void, boost::_mfi::mf1<void, SegmentationAction, boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> const&>, boost::_bi::list2<boost::_bi::value<SegmentationAction*>, boost::arg<1> > >::operator()<boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> > (this=0x7fff06c46bd8, a1=...) at /usr/include/boost/bind/bind_template.hpp:47
        a = {<boost::_bi::storage1<boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> const&>> = {a1_ = @0x7f663e415bc0}, <No data fields>}
#9  0x0000000000499955 in boost::detail::function::void_function_obj_invoker1<boost::_bi::bind_t<void, boost::_mfi::mf1<void, SegmentationAction, boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> const&>, boost::_bi::list2<boost::_bi::value<SegmentationAction*>, boost::arg<1> > >, void, boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> const&>::invoke (function_obj_ptr=..., a0=...) at /usr/include/boost/function/function_template.hpp:159
        f = 0x7fff06c46bd8
#10 0x0000000000496a26 in boost::function1<void, boost::shared_ptr<noether_msgs::SegmentGoal_<std::allocator<void> > const> const&>::operator() (this=0x7fff06c46bd0, a0=...) at /usr/include/boost/function/function_template.hpp:773
No locals.
#11 0x000000000048ce2e in actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > >::executeLoop (this=0x7fff06c46a18) at /opt/ros/kinetic/include/actionlib/server/simple_action_server_imp.h:384
        unlocker = {m = @0x7f663e415bb0, mtx = 0x7fff06c46b68}
        goal = {px = 0x1f91458, pn = {pi_ = 0x7f66280009d0}}
        lock = {m = 0x0, is_locked = false}
        __PRETTY_FUNCTION__ = "void actionlib::SimpleActionServer<ActionSpec>::executeLoop() [with ActionSpec = noether_msgs::SegmentAction_<std::allocator<void> >]"
        loop_duration = {<ros::DurationBase<ros::Duration>> = {sec = 0, nsec = 100000000}, <No data fields>}
#12 0x00000000004b643b in boost::_mfi::mf0<void, actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > > >::operator() (this=0x1f8b608, p=0x7fff06c46a18) at /usr/include/boost/bind/mem_fn_template.hpp:49
No locals.
#13 0x00000000004b5726 in boost::_bi::list1<boost::_bi::value<actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > >*> >::operator()<boost::_mfi::mf0<void, actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > > >, boost::_bi::list0> (this=0x1f8b618, f=..., a=...) at /usr/include/boost/bind/bind.hpp:253
No locals.
#14 0x00000000004b2b62 in boost::_bi::bind_t<void, boost::_mfi::mf0<void, actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > > >, boost::_bi::list1<boost::_bi::value<actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > >*> > >::operator() (this=0x1f8b608) at /usr/include/boost/bind/bind_template.hpp:20
        a = {<No data fields>}
#15 0x00000000004b0cf8 in boost::detail::thread_data<boost::_bi::bind_t<void, boost::_mfi::mf0<void, actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > > >, boost::_bi::list1<boost::_bi::value<actionlib::SimpleActionServer<noether_msgs::SegmentAction_<std::allocator<void> > >*> > > >::run (this=0x1f8b450) at /usr/include/boost/thread/detail/thread.hpp:116
No locals.
#16 0x00007f6656ad85d5 in ?? () from /usr/lib/x86_64-linux-gnu/libboost_thread.so.1.58.0
No symbol table info available.
#17 0x00007f66568b16ba in start_thread (arg=0x7f663e416700) at pthread_create.c:333
        __res = <optimized out>
        pd = 0x7f663e416700
        now = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140077107865344, -4939914349847261399, 0, 140733306921583, 140077107866048, 0, 5025342706377805609, 5025255241789329193}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
        pagesize_m1 = <optimized out>
        sp = <optimized out>
        freesize = <optimized out>
        __PRETTY_FUNCTION__ = "start_thread"
#18 0x00007f665604341d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
No locals.

```

**Comment by mpowelson** [comment]
Perhaps this occurs when we try to GetCell a deleted cell. It looks like when you delete a cell in a PolyData type, they get stored in VTK_EMPTY_CELL. I am just basing this on a similar issue found [HERE](https://gitlab.kitware.com/vtk/vtk/issues/16852)

**Comment by mpowelson** [comment]
I think I finally fixed this issue. I believe I was assigning meaningless cell ID's for the last mesh. This was resolved in commit 8066c78b2def6840527c6b16d0e955877fdcc30a
